client/packages/system/src/Immunisation/ImmunisationProgramView/VaccineCreateModal.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { FC, useState } from 'react';
+import {
+  useDialog,
+  Grid,
+  DialogButton,
+  useTranslation,
+  FnUtils,
+  InlineSpinner,
+  BasicTextInput,
+  Box,
+  InputLabel,
+  Autocomplete,
+} from '@openmsupply-client/common';
+
+interface VaccineCreateModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const createNewProgram = (seed?: any | null): any => ({
+  id: FnUtils.generateUUID(),
+  name: '',
+  demographicGroup: '',
+  coverageRate: '',
+  calculatedDemand: '',
+  ...seed,
+});
```

Comment:
I'm thinking it might be more consistent to just put the name in the create modal.
Then we can do all the other updates on the Detail Edit screen, Keeps the API more focussed too.
This is what I'm doing for programs...